libs/network/socketserver.c

```diff
@@ -352,9 +366,25 @@ bool socketServerConnectionsIdle(JsNetwork *net) {
         jswrap_stream_pushData(connection, receiveData, true);
       }
       jsvUnLock(receiveData);
+
+      // fire the error listeners
+      bool hadError = error < 0 && error != SOCKET_ERR_CLOSED;
+      JsVar *params[1];
+      if (hadError) {
+        params[0] = jsvNewWithFlags(JSV_OBJECT);
```

Comment:
Could this be a separate function? It'd reduce duplication with the other Error code below, and would make it easier to tweak later on to print an error message if there was no handler.